integrated_channels/api/v1/logs/views.py

```diff
@@ -24,32 +25,83 @@
     LearnerDataTransmissionAudit,
 )
 
-from .serializers import ContentSyncStatusSerializer, GenericLearnerSyncStatusSerializer, LearnerSyncStatusSerializer
+from .serializers import ContentSyncStatusSerializer, LearnerSyncStatusSerializer
 
 LOGGER = logging.getLogger(__name__)
 
 
+class ReportingSyncStatusPagination(PageNumberPagination):
+    page_size = 10
+
+    def get_paginated_response(self, data):
+        return Response(OrderedDict([
+            ('pages_count', self.page.paginator.num_pages),
+            ('count', self.page.paginator.count),
+            ('next', self.get_next_link()),
+            ('previous', self.get_previous_link()),
+            ('results', data)
+        ]))
+
+
+class InvalidSortByParamException(exceptions.APIException):
+    status_code = 400
+    default_detail = "Invalid sort_by filter."
+    default_code = "validation_error"
+
+
+class InvalidChannelCodeException(exceptions.APIException):
+    status_code = 400
+    default_detail = "Invalid channel code."
+    default_code = "validation_error"
+
+
 class ContentSyncStatusViewSet(PermissionRequiredForIntegratedChannelMixin, viewsets.ModelViewSet):
     """
     Sync-status APIs for `ContentMetadataItemTransmission` items.
     """
     serializer_class = ContentSyncStatusSerializer
     permission_classes = (permissions.IsAuthenticated,)
     permission_required = 'enterprise.can_access_admin_dashboard'
-    pagination_class = PageNumberPagination
+    pagination_class = ReportingSyncStatusPagination
 
-    def get_queryset(self):
-        enterprise_customer_uuid = self.kwargs.get('enterprise_customer_uuid', None)
-        integrated_channel_code = self.kwargs.get('integrated_channel_code', None)
-        plugin_configuration_id = self.kwargs.get('plugin_configuration_id', None)
+    def validate(self):
+        integrated_channel_code = self.kwargs.get('integrated_channel_code')
         channel_config_cls = EnterpriseCustomerPluginConfiguration.get_class_by_channel_code(integrated_channel_code)
         if channel_config_cls is None:
-            raise exceptions.ParseError("Invalid channel code.")
+            raise InvalidChannelCodeException
+
+        if sort_by_param := self.request.GET.get('sort_by'):
+            if sort_by_param.replace("-", "") not in ["sync_status", "sync_last_attempted_at"]:
+                if sort_by_param.replace("-", "") not in [field.name for field in ContentMetadataItemTransmission._meta.fields]:
+                    raise InvalidSortByParamException
+
+    def get_queryset(self):
+        self.validate()
+        content_metadata_transmission_audit_filter = {
+            "enterprise_customer_id": self.kwargs.get('enterprise_customer_uuid'),
+            "integrated_channel_code": self.kwargs.get('integrated_channel_code'),
+            "plugin_configuration_id": self.kwargs.get('plugin_configuration_id'),
+        }
```

Comment:
Are we just picking out a few keys from a dictionary here?  If so, we could just do
```suggestion
        content_metadata_fields = ('enterprise_customer_uuid', 'integrated_channel_code', 'plugin_configuration_id')
        content_metadata_transmission_audit_filter = {k: self.kwargs.get(k) for k in content_metadata_fields}
```
If we do things like this a lot, it's a good candidate for making into a helper method, or using library like [pydash.pick](https://pydash.readthedocs.io/en/latest/api.html?highlight=pick#pydash.objects.pick)

Reply:
to my knowledge we're not doing this very frequently, but I do like your suggestion!